skyportal/handlers/api/thumbnail.py

```diff
@@ -3,11 +3,46 @@
 import base64
 from pathlib import Path
 from marshmallow.exceptions import ValidationError
+import sqlalchemy as sa
 from sqlalchemy.exc import StatementError
 from PIL import Image, UnidentifiedImageError
+
 from baselayer.app.access import permissions, auth_or_token
 from ..base import BaseHandler
-from ...models import Obj, Thumbnail
+from ...models import Obj, Thumbnail, User
+
+
+def post_thumbnail(data, user_id, session):
+    """Post source to database.
+    data: dict
+        Thumbnail dictionary
+    user_id : int
+        SkyPortal ID of User posting the Thumbnail
+    session: sqlalchemy.Session
+        Database session for this transaction
+    """
+
+    user = session.scalar(sa.select(User).where(User.id == user_id))
+
+    obj = session.scalars(Obj.select(user).where(Obj.id == data["obj_id"])).first()
+
+    if obj is None:
+        raise AttributeError(f"Invalid obj_id: {data['obj_id']}")
+
+    try:
+        t = create_thumbnail(data['data'], data['ttype'], data["obj_id"], session)
```

Comment:
It looks like `create_thumbnail` does something really similar to this function. Any chance we can combine the two into one function and make less spaghetti?

Reply:
Good point, I combined these two.